### DISCUSSION TITLE
How to use special characters in my own shortcuts?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#413] [Ask a question] [answered]
I'm currently writing a CSV file parser for a personal wiki project. The shortcut to display a file is: 

    [csv $filename $delimiter $class]

The delimiter parameter only works for `;` semicolon and `,` comma-separated files, but not for tab-separated csv strings. I tried passing the tab character as `\t`, but Yellow seems to ignore this. Can I somehow use these characters in a shortcut?

Steffen

**Comment by GiovanniSalmeri** [comment]
I do not know if this is possible with a shortcut, but I am using for an extension the PHP functions `fgetcsv`, `str_getcsv` ecc., where it is possible to specify any delimiter, enclosure character and escape character. Perhaps you could use them without rewriting a parser?

**Comment by markseuffert** [comment, marked as answer]
Try to convert your argument, e.g. `$delimiter = strreplaceu("\\t", "\t", $delimiter);`

**Comment by schulle4u** [comment]
Ah... thanks Mark, exactly what I needed. Works now. :)

@GiovanniSalmeri Yes, I'm already using these functions. The shortcut only should provide an easy way to load any CSV file into a page. I'll upload my code when it is stable enough for the public. 

Steffen
